src/test/java/org/junit/platform/engine/test/event/ExecutionEventConditions.java

```diff
@@ -0,0 +1,156 @@
+	}
+
+	public static Condition<ExecutionEvent> container(Class<?> clazz) {
+		return container(clazz.getName());
+	}
+
+	public static Condition<ExecutionEvent> container(String uniqueIdSubstring) {
+		return allOf(container(), uniqueIdSubstring(uniqueIdSubstring));
+	}
+
+	public static Condition<ExecutionEvent> container() {
+		return new Condition<>(byTestDescriptor(TestDescriptor::isContainer), "is a container");
+	}
+
+	public static Condition<ExecutionEvent> dynamicTestRegistered(String uniqueIdSubstring) {
+		return allOf(type(DYNAMIC_TEST_REGISTERED), uniqueIdSubstring(uniqueIdSubstring));
+	}
+
+	public static Condition<ExecutionEvent> uniqueIdSubstring(String uniqueIdSubstring) {
+		return new Condition<>(
+			byTestDescriptor(where(testDescriptor -> testDescriptor.getUniqueId().toString(),
+				uniqueId -> uniqueId.contains(uniqueIdSubstring))),
+			"descriptor with uniqueId substring \"%s\"", uniqueIdSubstring);
+	}
+
+	public static Condition<ExecutionEvent> displayName(String displayName) {
+		return new Condition<>(byTestDescriptor(where(TestDescriptor::getDisplayName, isEqual(displayName))),
+			"descriptor with display name \"%s\"", displayName);
+	}
+
+	public static Condition<ExecutionEvent> skippedWithReason(String expectedReason) {
+		return allOf(type(SKIPPED), reason(expectedReason));
+	}
+
+	public static Condition<ExecutionEvent> started() {
+		return type(STARTED);
+	}
+
+	public static Condition<ExecutionEvent> abortedWithReason(Condition<? super Throwable> causeCondition) {
+		return finishedWithCause(ABORTED, causeCondition);
+	}
+
+	public static Condition<ExecutionEvent> finishedWithFailure(Condition<? super Throwable> causeCondition) {
+		return finishedWithCause(FAILED, causeCondition);
+	}
+
+	private static Condition<ExecutionEvent> finishedWithCause(Status expectedStatus,
+			Condition<? super Throwable> causeCondition) {
+		return finished(allOf(status(expectedStatus), cause(causeCondition)));
+	}
+
+	public static Condition<ExecutionEvent> finishedWithFailure() {
+		return finished(status(FAILED));
+	}
+
+	public static Condition<ExecutionEvent> finishedSuccessfully() {
+		return finished(status(SUCCESSFUL));
+	}
+
+	public static Condition<ExecutionEvent> finished(Condition<TestExecutionResult> resultCondition) {
+		return allOf(type(FINISHED), result(resultCondition));
+	}
+
+	public static Condition<ExecutionEvent> type(Type expectedType) {
+		return new Condition<>(byType(expectedType), "type is %s", expectedType);
+	}
+
+	public static Condition<ExecutionEvent> result(Condition<TestExecutionResult> condition) {
+		return new Condition<>(byPayload(TestExecutionResult.class, condition::matches), "event with result where %s",
+			condition);
+	}
+
+	public static Condition<ExecutionEvent> reason(String expectedReason) {
+		return new Condition<>(byPayload(String.class, isEqual(expectedReason)), "event with reason '%s'",
+			expectedReason);
+	}
+
+}
```

Comment:
Have you considered checking the nullness of the parameters of all the methods in this class, using `Objects.requireNonNull` or Guava's `Preconditions.checkNotNull`?